maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -305,25 +305,70 @@ public void promptForJavaHomeIfMissing(Server server) {
 
         if (interactiveMode) {
             List<String> paths = new ArrayList<>();
-            paths.add("JAVA_HOME (currently: " + System.getenv("JAVA_HOME") + ")");
+            paths.add("JAVA_HOME (currently: " + System.getProperty("java.home") + ")");
             paths.addAll(getJavaHomeOptions());
 
             String path = promptForMissingValueWithOptions(SDKConstants.OPENMRS_SDK_JDK_OPTION,
                     server.getJavaHome(), "path", paths, SDKConstants.OPENMRS_SDK_JDK_CUSTOM, null);
 
+            String requiredJdkVersion;
+            if (server.getPlatformVersion().charAt(0) == 1) {
+                requiredJdkVersion = "1.7";
+            }
+            else {
+                requiredJdkVersion = "1.8";
+            }
+
+            // Use default JAVA_HOME
             if (path.equals(paths.get(0))) {
-                // Use default JAVA_HOME
-                server.setJavaHome(null);
+                if (System.getProperty("java.version").startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(null);
+                }
+                else if (System.getProperty("java.version").startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(null);
+                }
+                else {
+                    showMessage("Your JAVA_HOME version doesn't fit platform requirements:");
+                    showMessage("JAVA_HOME version: " + System.getProperty("java.version"));
+                    showMessage("Required: " + requiredJdkVersion);
+                    promptForJavaHomeIfMissing(server);
+                }
             } else if (!isJavaHomeValid(path)) {
                 System.out.println(SDKConstants.OPENMRS_SDK_JDK_CUSTOM_INVALID);
                 promptForJavaHomeIfMissing(server);
             } else {
-                server.setJavaHome(path);
-                addJavaHomeToSdkProperties(path);
+                String jdkUnderSpecifiedPathVersion = extractJavaVersionFromPath(path);
+                if (jdkUnderSpecifiedPathVersion.startsWith(requiredJdkVersion)) {
+                    server.setJavaHome(path);
+                    addJavaHomeToSdkProperties(path);
+                }
+                else {
+                    showMessage("JDK in custom path (" + path + ") doesn't match platform requirements:");
+                    showMessage("JDK version: " + jdkUnderSpecifiedPathVersion);
+                    showMessage("Required: " + requiredJdkVersion);
+                    promptForJavaHomeIfMissing(server);
+                }
             }
         }
+    }
 
-
+    private String extractJavaVersionFromPath(String path) {
+        path = path.substring(0, path.length()-4);
+        File jdkReleasePropertiesFile = new File(path, "release");
```

Comment:
Thats how JDK version is extracted from JDK directory.
Works for Oracle releases.